src/modules/compliance/src/lib/procedures/ensureSysctl.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <CommonUtils.h>
+#include <Evaluator.h>
+#include <Regex.h>
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+namespace compliance
+{
+
+AUDIT_FN(ensureSysctl, "sysctlName:Name of the kernel module:M", "value:Regex value Sysctl that needst to be match",
+    "test_procfs:Prefix for the /proc/sys from where sysctl will be read", "test_systemd_sysctl:Path where systemd-sysctl tool is installed")
+{
+    char* output = NULL;
+    std::string procfsLocation = "/proc/sys";
+    std::string systemdSysctl = "/lib/systemd/systemd-sysctl";
+
+    auto it = args.find("sysctlName");
+    if (it == args.end())
+    {
+        return Error("Missing 'sysctlName' parameter", EINVAL);
+    }
+    auto sysctlName = std::move(it->second);
+
+    it = args.find("value");
+    if (it == args.end())
+    {
+        return Error("Missing 'value' parameter", EINVAL);
+    }
+    auto sysctlValue = std::move(it->second);
+
+    if (args.find("test_procfs") != args.end())
+    {
+        procfsLocation = args["test_procfs"];
+    }
+    if (args.find("test_systemd_sysctl") != args.end())
+    {
+        systemdSysctl = args["test_systemd_sysctl"];
+    }
+
+    auto sysctlPath = sysctlName;
+    std::replace(sysctlPath.begin(), sysctlPath.end(), '.', '/');
+    std::string procSysPath = procfsLocation + std::string("/") + sysctlPath;
+
+    output = LoadStringFromFile(procSysPath.c_str(), false, log);
+    if (output == NULL)
+    {
+        logstream << "Fail sysctl: '" << sysctlName << "' NOT compliant. Expected value: '" << sysctlValue << "' got NULL";
+        return false;
+    }
+    std::string sysctlOutput(output);
+    FREE_MEMORY(output);
+
+    // remove newline caracter
+    if (*sysctlOutput.rbegin() == '\n')
+    {
+        sysctlOutput.erase(sysctlOutput.size() - 1);
+    }
+
+    regex valueRegex;
+    try
+    {
+        valueRegex = regex(sysctlValue);
+    }
+    catch (const std::exception& e)
+    {
+        OsConfigLogError(nullptr, "Regex error: %s", e.what());
+        return false;
```

Comment:
I'd actually return an Error here, since that's what it is, and we will have proper handling of those in the interface (e.g. won't stop the whole benchmark).